tests/scenarios/v2-addon-dev-watch-test.ts

```diff
@@ -134,6 +137,139 @@ Scenarios.fromProject(() => baseV2Addon())
+              // to guess for how long it'll take for the file system to update our file.
+              //
+              // the `stat` is measured in `ms`, so it's still pretty fast
+              await fs.writeFile(demoHbs, someContent + `\n`);
+              await aBit(10);
+              await watcher?.nextBuild();
+            },
+          });
+
+          // removing hbs modifies colocated js to not import hbs anymore
+          await becomesModified({
+            filePath: distPathDemoComp,
+            assert,
+            // Update a component
+            fn: async () => {
+              // generally it's bad to introduce time dependencies to a test, but we need to wait long enough
+              // to guess for how long it'll take for the file system to update our file.
+              //
+              // the `stat` is measured in `ms`, so it's still pretty fast
+              await fs.rm(demoHbs);
+              await aBit(10);
+              await watcher?.nextBuild();
+            },
+          });
+
+          await fs.writeFile(demoHbs, demoContent);
+          await watcher?.nextBuild();
+
+          // updating hbs content should not update unrelated files
+          await isNotModified({
+            filePath: distPath,
+            assert,
+            // Update a component
+            fn: async () => {
+              let someContent = await fs.readFile(demoHbs);
+
+              // generally it's bad to introduce time dependencies to a test, but we need to wait long enough
+              // to guess for how long it'll take for the file system to update our file.
+              //
+              // the `stat` is measured in `ms`, so it's still pretty fast
+              await fs.writeFile(demoHbs, someContent + `\n\n`);
+              await aBit(10);
+              await watcher?.nextBuild();
+            },
+          });
+
+          // updating hbs content should update resulting app re-exported component
+          distPath = path.join(addon.dir, 'dist/_app_/components/test.js');
+          await isNotModified({
+            filePath: distPath,
+            assert,
+            // Update a component
+            fn: async () => {
+              let someContent = await fs.readFile(demoHbs);
+
+              // generally it's bad to introduce time dependencies to a test, but we need to wait long enough
+              // to guess for how long it'll take for the file system to update our file.
+              //
+              // the `stat` is measured in `ms`, so it's still pretty fast
+              await fs.writeFile(demoHbs, someContent + `\n`);
+              await aBit(10);
+              await watcher?.nextBuild();
+            },
+          });
+
+          // updating template only hbs should update the dist output
+          distPath = path.join(addon.dir, 'dist/components/button.js');
+          await isNotModified({
```

Comment:
why is this _not_ getting modified?

Reply:
that's actually correct. The rexport file did not change.
Wrong comment

Reply:
it should update---